app/models/mentor/application.rb

```diff
@@ -0,0 +1,118 @@
+module Mentor
+  class Application
+    include ActiveModel::Model
+
+    DB = ActiveRecord::Base.connection
+
+    attr_accessor :id, :team_name
+    attr_accessor :project_id, :project_name
+    attr_accessor :project_plan, :why_selected_project
+    attr_accessor :student0, :student1
+    attr_accessor :first_choice
+
+    def choice
+      first_choice ? 1 : 2
+    end
+
+    def student0=(attrs)
+      attrs     = JSON.parse(attrs).instance_eval(&studentize)
```

Comment:
Yeah, I agree with the inside-out brackets (if only we had a pipe-operator in Ruby … :wink: )

I'm just unsure if binding the return value of higher order functions/methods to methods as their block parameter which in turn `#tap`s the (implicit) block arg isn't something anyone with less than 4-5 years of Ruby experience immediately shies away from.

I don't know what weighs more so I'll totally shove the responsibility to make a decision back to you :trollface: 

As for the mutation: I don't think memory usage will be an issue for the Teams App anytime soon. Otoh, I can foresee a very nasty debugging session when we try to figure out where certain keys in the passed `params` hash went to. But I don't have strong feelings here, either. You decide! :)